packages/@azure/eventhubs/client/lib/linkEntity.ts

```diff
@@ -168,11 +168,18 @@ export class LinkEntity {
     clearTimeout(this._tokenRenewalTimer as NodeJS.Timer);
     if (link) {
       try {
-        // This should take care of closing the link and it's underlying session. This should also
+        if (link.isOpen()){
+        // Closing the link and its underlying session if the link is open. This should also
         // remove them from the internal map.
         await link.close();
         log.link("[%s] %s '%s' with address '%s' closed.", this._context.connectionId, this._type,
           this.name, this.address);
+        }else{
+        //  The link is already closed. Removing it and it's underlying session from the internal map and releasing its listeners.
+          await link.remove();
```

Comment:
My last concern here is if we are ever in a situation where we want to remove the link, but not the session....

Can you figure out if we create multiple links using the same session? Or use different sessions for different links?

Reply:
When we create an amqp link. It either uses the provided session or creates a new one. It means that we can provide the same session for multiple links.

Reply:
As discussed offline, lets hold off on removing the rhea link which internally forcefully removes not just the link but the session itself.

From our side, lets just remove all the listeners from our reciever object inside the `detached` call. This should fix the problem for our users.

Independently, lets work on a sample that uses rhea-promise directly to replicate this problem and then log an issue in the rhea repo

Reply:
>From our side, lets just remove all the listeners from our reciever object inside the detached call. This should fix the problem for our users

It will not fix our problem, if we remove all our listeners from our receiver object, we are not able to pick up receiving at whatever state the queue was left in, instead we get `Something is busted` error but if we call `link.remove()` method then everything works as expected.

As discussed offline we are not reopening the session in our SDK, so I think now we can use `link.remove()` but let me know if you have any other workaround.


>Independently, lets work on a sample that uses rhea-promise directly to replicate this problem and then log an issue in the rhea repo

Ok, I'll work on a sample